File: app/src/main/java/com/vgaidarji/cimatters/LoginPresenter.java
package com.vgaidarji.cimatters;

class LoginPresenter {

    private static final String EMAIL = "test@test.com";
    private static final String PASSWORD = "1111";
    private static final String WRONG_CREDENTIALS = "Wrong credentials.";
    private final LoginView view;

    LoginPresenter(LoginView view) {
        this.view = view;
    }

    void onLoginClick(String email, String password) {
        if (email.equals(EMAIL) && password.equals(PASSWORD)) {
            view.openNextActivity();
        } else {
            view.showError(WRONG_CREDENTIALS);
        }
    }
}

File: app/src/main/java/com/vgaidarji/cimatters/LoginPresenter.kt
package com.vgaidarji.cimatters

import android.util.Log

internal class LoginPresenter(private val view: LoginView) {
    fun onLoginClick(email: String, password: String) {
        Log.d("test123","email + ${email}, password + ${password}, ${email == EMAIL && password == PASSWORD}")
        if (email == EMAIL && password == PASSWORD) {
            view.openNextActivity()
        } else {
            view.showError(WRONG_CREDENTIALS)
        }
    }

    companion object {
        private const val EMAIL = "test@test.com"
        private const val PASSWORD = "1111"
        private const val WRONG_CREDENTIALS = "Wrong credentials."
    }
}
